modules/metadata-support/src/main/java/org/mule/runtime/metadata/api/cache/ConfigurationMetadataCacheIdGenerator.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.metadata.api.cache;
+
+import org.mule.runtime.ast.api.ComponentAst;
+
+import java.util.Optional;
+
+/**
+ * Allows to retrieve the {@link MetadataCacheId} for Configs. It's used since the
+ * {@link org.mule.runtime.extension.api.component.ComponentParameterization} has no information of the configs, including its
+ * child elements.
+ *
+ * @since 4.5
+ */
+public interface ConfigurationMetadataCacheIdGenerator {
+
+  /**
+   * Returns the {@link MetadataCacheId} corresponding to the configuration.
+   * 
+   * @param configName      - the configuration name to get the {@link MetadataCacheId}
+   * @param asGlobalElement - if true, it will return the id corresponding to the calculation of this config as a global element.
+   *                        If false, it will return the id corresponding to a referenced value.
+   */
+  Optional<MetadataCacheId> getConfigMetadataCacheId(String configName, boolean asGlobalElement);
+
+  /**
+   * Adds the config to be considered when getting the Cache Ids. If the config was already set before, the values will be
+   * recalculated.
+   */
+  void addConfiguration(ComponentAst configAst);
```

Comment:
is this something the user of this has to be aware of? it sound like this belongs to an internal interface rather than here...

Reply:
It is to add the config, when the ast is available. Check https://github.com/mulesoft/mule-framework/pull/114/files#diff-6256e720b7e3f80bf26ba49109b8b9fea50b14ab581c56ccbe4a792284b81c40R73